mkosi/gentoo.py

```diff
@@ -0,0 +1,549 @@
+        sudo install --owner=portage --group=portage --mode=0755 --directory /var/cache/binpkgs
+
+        sudo tee /etc/portage/repos.conf/eselect-repo.conf > /dev/null <<- EOF
+        [gentoo]
+        location = /var/db/repos/gentoo
+        sync-type = git
+        sync-uri = https://anongit.gentoo.org/git/repo/gentoo.git
+        EOF
+
+        git clone https://anongit.gentoo.org/git/proj/portage.git --depth=1
+        cd portage
+        tee setup.cfg > /dev/null <<- EOF
+        [build_ext]
+        portage-ext-modules=true
+        EOF
+
+        python setup.py build_ext --inplace --portage-ext-modules
+
+        sudo python setup.py install
+
+        sudo ln -s --relative /var/db/repos/gentoo/profiles/default/linux/amd64/17.1/no-multilib/systemd /etc/portage/make.profile
+        """
+        try:
+            from portage.const import (  # type: ignore
+                CUSTOM_PROFILE_PATH,
+                EBUILD_SH_ENV_DIR,
+                PROFILE_PATH,
+                USER_CONFIG_PATH,
+            )
+        except ImportError as e:
+            MkosiPrinter.warn(NEED_PORTAGE_MSG)
+            MkosiPrinter.info(PORTAGE_INSTALL_INSTRUCTIONS)
+            raise MkosiException(e)
+
+        return dict(profile_path=PROFILE_PATH,
+                    custom_profile_path=CUSTOM_PROFILE_PATH,
+                    ebuild_sh_env_dir=EBUILD_SH_ENV_DIR,
+                    portage_cfg_dir=USER_CONFIG_PATH)
+
+    def __init__(
+        self,
+        args: CommandLineArguments,
+        root: Path,
+        do_run_build_script: bool,
+    ) -> None:
+
+        ret = self.try_import_portage()
+
+        from portage.package.ebuild.config import config  # type: ignore
+
+        self.portage_cfg = config(config_root=str(root), target_root=str(root),
+                                  sysroot=str(root), eprefix=None)
+
+        PORTAGE_MISCONFIGURED_MSG = "You have portage(5) installed but it's probably missing defaults, bailing out"
+        # we check for PORTDIR, but we could check for any other one
+        if self.portage_cfg['PORTDIR'] is None:
+            die(PORTAGE_MISCONFIGURED_MSG)
+
+        if "build-script" in ARG_DEBUG:
+            for k, v in self.portage_cfg.items():
+                print(f"{k} = {v}")
+
+        self.profile_path = root / ret["profile_path"]
+        self.custom_profile_path = root / ret["custom_profile_path"]
+        self.ebuild_sh_env_dir = root / ret["ebuild_sh_env_dir"]
+        self.portage_cfg_dir = root / ret["portage_cfg_dir"]
+
+        self.portage_cfg_dir.mkdir(parents=True, exist_ok=True)
+
+        self.DEFAULT_NSPAWN_PARAMS = [
+            "--capability=CAP_SYS_ADMIN,CAP_MKNOD",
+            "--tmpfs=/sys",
+            f"--bind={self.portage_cfg['PORTDIR']}",
+            f"--bind={self.portage_cfg['DISTDIR']}",
+            f"--bind={self.portage_cfg['PKGDIR']}",
+        ]
+
+        jobs = multiprocessing.cpu_count()
```

Comment:
```suggestion
        jobs = os.cpu_count() or 1
```
is I think rather what you want. I always perk up when I see `multiprocessing` imported, since I've seen too many users do weird things with it. `multiprocessing.cpu_count` is only a thin wrapper around `os.cpu_count` and the latter cannot blow up.